File: common/src/main/java/dev/qixils/aviary/common/messaging/serializers/SingletonByteSerializer.java
package dev.qixils.aviary.common.messaging.serializers;

import org.jetbrains.annotations.NotNull;

/**
 * An extension of {@link ByteSerializer} which also supports deserializing from a single byte.
 */
public interface SingletonByteSerializer<T> extends ByteSerializer<T> {

	/**
	 * Converts a single byte to an object.
	 *
	 * @param data the byte
	 * @return the object
	 */
	@NotNull
	T deserialize(byte data);
}

File: common/src/main/kotlin/dev/qixils/aviary/common/messaging/serializers/SingletonByteSerializer.kt
package dev.qixils.aviary.common.messaging.serializers

/**
 * An extension of [ByteSerializer] which also supports deserializing from a single byte.
 */
interface SingletonByteSerializer<T : Any> : ByteSerializer<T> {

	/**
	 * Converts a single byte to an object.
	 *
	 * @param data the byte
	 * @return the object
	 */
	fun deserialize(data: Byte): T
}
